lib/biokbase/catalog/db.py

```diff
@@ -120,130 +123,217 @@ class MongoCatalogDBI:
 
     def __init__(self, mongo_host, mongo_db, mongo_user, mongo_psswd, mongo_authMechanism):
 
-        # create the client
-        self.mongo = MongoClient('mongodb://' + mongo_host)
+        self.mongo_host = mongo_host
+        self.mongo_db = mongo_db
+        self.mongo_user = mongo_user
+        self.mongo_psswd = mongo_psswd
+        self.mongo_authMechanism = mongo_authMechanism
+
+        # MongoDB client and database initialization deferred
+        self._mongo_client_initialized = False
+        self.mongo = None
+        self.db = None
+        self.index_created = defaultdict(int)
```

Comment:
Should this be a bool?

Reply:
I used 1 for True and 0 for False.

Reply:
Yes, so why not use a defaultdict(bool) instead?

Reply:
👍